js/sketchfab.js

```diff
@@ -119,7 +120,13 @@ SketchfabPlugin.prototype.onSubmit = function onSubmit() {
 	var modelId = url.substring(url.lastIndexOf('/') + 1);
 	var text = this.renderText(modelId);
 
-	this.insertAtCursor(text);
+	if ((this.mode || '').toLowerCase() === 'wysiwyg') {
```

Comment:
Yeah I've seen those ; the thing is that the variable editor is an entierly new instance that is inaccessible outside of the code block in which it is used (that is, the editor plugin's code). For me to be able tu use it, it'd need to be exposed in some way. I could not find a way to retrieve the instance given the existing code / the way the lib behaves (see https://github.com/xing/wysihtml5/issues/155 ; it's an old post but I could not find more recent and relevant content).

Given that using the data-attr seems really bad, I've switched gears and I'm now listening :
$(document).on('editorParseRules')

Which seems to be a custom event that is fired by the editor plugin once it has loaded the proper scripts for wysiwyg mode. What do you think about this behaviour ?